From e9dcad4aa3ae92c8b5f8905653950ee39aec28a0 Mon Sep 17 00:00:00 2001
From: Dean Sanner <dsanner@us.ibm.com>
Date: Wed, 30 Aug 2017 06:07:49 -0500
Subject: [PATCH v1 1005/1006] Fix up GPE0/1 reset logic

Previous commit to remove logic checking for halt state
went too far and removed scom address setup.  Adding that
part back in

Change-Id: I283a481b16f78f22f8ccfc0bf8d8d26b2f88f690
---
 .../chips/p9/procedures/hwp/pm/p9_pm_occ_gpe_init.C     | 17 +++++++++++++++++
 1 file changed, 17 insertions(+)

diff --git a/src/import/chips/p9/procedures/hwp/pm/p9_pm_occ_gpe_init.C b/src/import/chips/p9/procedures/hwp/pm/p9_pm_occ_gpe_init.C
index 912d5d6..97e8b13 100644
--- a/src/import/chips/p9/procedures/hwp/pm/p9_pm_occ_gpe_init.C
+++ b/src/import/chips/p9/procedures/hwp/pm/p9_pm_occ_gpe_init.C
@@ -144,6 +144,23 @@ fapi2::ReturnCode pm_occ_gpe_reset(
 
     do
     {
+        if (i_engine == p9occgpe::GPE0)
+        {
+            l_controlReg    =   PU_GPE0_PPE_XIXCR;
+            l_statusReg     =   PU_GPE0_GPEXIXSR_SCOM;
+            l_instrAddrReg  =   PU_GPE0_PPE_XIDBGPRO;
+            l_intVecReg     =   PU_GPE0_GPEIVPR_SCOM;
+            l_gpeBaseAddress.push_back( GPE0_BASE_ADDRESS );
+        }
+        else if (i_engine == p9occgpe::GPE1)
+        {
+            l_controlReg = PU_GPE1_PPE_XIXCR;
+            l_statusReg = PU_GPE1_GPEXIXSR_SCOM;
+            l_instrAddrReg = PU_GPE1_PPE_XIDBGPRO;
+            l_intVecReg = PU_GPE1_GPEIVPR_SCOM;
+            l_gpeBaseAddress.push_back( GPE1_BASE_ADDRESS );
+        }
+
         // Halt the OCC GPE
         l_data64.flush<0>().insertFromRight(p9hcd::HALT, 1, 3);
         FAPI_TRY(putScom(i_target, l_controlReg, l_data64),
-- 
1.8.2.2

